features/323-login-id/design.md

```diff
@@ -0,0 +1,109 @@
+# Background
+
+## Current Design
+
+Under current design of authentication:
+
+1. Each **User** is associated with some **Principal**.
+3. Each **Principal** is associated with one **Identity Provider** (e.g. password authentication, OAuth).
+4. Password authentication principals is represented by tuple (**Login ID Key**, **Login ID**, password hash).
+5. App developer specifies a list of white-listed login ID key.
+6. To signup using password authentication, user can provide a list of tuple (login ID key, login ID). e.g. `{ "username": "test", "email": "a@example.com" }`
+7. For password authentication principals, user can login using a tuple (login ID key, login ID). e.g. `{ "email": "a@example.com" }`
```

Comment:
login ID Key is not a must. User can login only with LoginID. As specified at [spec 294](https://github.com/SkygearIO/features/blob/master/features/294-login-without-specifying-login-id-key/api.md)

Reply:
@kiootic please update this to reflect the current design